src/microstate.js

```diff
@@ -1,12 +1,20 @@
-import { map } from "funcadelic";
+import { map, append } from "funcadelic";
 import analyze, { collapseState } from "./structure";
 import { keep, reveal } from "./utils/secret";
 import SymbolObservable from "symbol-observable";
+import thunk from './thunk';
 
 export default class Microstate {
   constructor(tree, value) {
     keep(this, { tree, value });
-    return map(transition => transition, this);
+    
+    let state = thunk(() => collapseState(tree, value));
```

Comment:
As of 0.3.2, `funcadelic` has a `stable` function, that we could potentially use. https://github.com/cowboyd/funcadelic.js/blob/master/src/stable.js#L1

Reply:
But rather than making a stable thunk here, I think we should just make the semigroup instance of `Object` stable. In other words we map the appended properties, and if it's a getter, then we transparently convert it to a stable thunk here https://github.com/cowboyd/funcadelic.js/blob/master/src/semigroup/object.js#L9.

That would make the append be a simple:

```js
return append(map(transition => transition, this), {
  get state() {
    return collapseState(tree, value));
  }
});

Reply:
I have everything you described in a branch of funcadelic. I’ll make a PR